sympy/polys/rings.py

```diff
@@ -2531,6 +2531,125 @@ def compose(f, x, a=None):
 
         return poly
 
+
+    def coeff_wrt(self, sym, dg):
+        """
+        Coefficient of ``f`` with respect to ``x**deg``.
+
+        Treating ``f`` as a univariate polynomial in ``x`` this finds the
+        coefficient of ``x**deg`` as a polynomial in the other generators.
+
+        Parameters
+        ==========
+
+        p : sympy.Poly
+            The polynomial to compute the sparse coefficient for.
+        sym : int
+            The symbol to compute the coefficient for.
+        dg : int
+            The degree of the monomial to compute the coefficient for.
+
+        Returns
+        =======
+
+        sympy.Number
+            The sparse coefficient of the polynomial at the given symbol and degree.
+
+        Examples
+        ========
+
+        >>> from sympy.polys import ring, ZZ
+        >>> _, x, y, z = ring("x, y, z", ZZ)
+        >>> p = 2*x**4 + 3*y**4 + 10*z**2 + 3
+        >>> dg = 2
+        >>> sym = 2
+        >>> coeff_wrt(p, sym, dg)
+        10
+
+        """
+        p = self
+        terms = [(m, c) for m, c in p.iterterms() if m[sym] == dg]
+        monoms, coeffs = zip(*terms)
+        monoms = [m[:sym] + (0,) + m[sym + 1:] for m in monoms]
+        return p.ring.from_dict(dict(zip(monoms, coeffs)))
+
+
+    def prem(self, g, x):
+        """
+        Computes the pseudo-remainder of the polynomial `f` with respect to `g`.
+
+        Parameters
+        ==========
+
+        f : PolyElement
+            The polynomial to compute the pseudo-remainder for.
+        g : PolyElement
+            The polynomial to divide `f` by.
+        x : Symbol
+            The main variable of the polynomials.
+
+        Returns
+        =======
+
+        PolyElement: The pseudo-remainder polynomial.
+
+        Raises
+        ======
+
+        ZeroDivisionError: If the degree of `g` is negative.
+        ValueError: If the algorithm encounters an unexpected condition.
+
+        Examples
+        ========
+
+        >>> from sympy.polys import ring, ZZ
+        >>> _, x, y, z = ring("x, y, z", ZZ)
+
+        >>> f = x**2 + x*y
+        >>> g = 2*x + 2
+        >>> prem(f, g, 0)
```

Comment:
Here also, It would be called with an instance of the class.

Reply:
https://github.com/sympy/sympy/blob/998a04f12053059be26a44d5b7011eda1def7929/sympy/polys/rings.py#L2659-L2660

Investigate the repeat of function name here. It is causing a flake8 error.